client/cypress/integration/product-list.spec.ts

```diff
@@ -246,7 +245,51 @@ describe('Product already in shopping list add button', () => {
 
   // Product already in shopping list
   it('should click the add button, then click the button to go to the shopping list page', () => {
-    page.clickExpansionAddShoppingButton('baking supplies');
+    const testNewProduct: Product = {
+      _id: 'testID',
+      productName: 'Frog Legs',
+      brand: 'Marsh  Inc.',
+      category: 'produce',
+      description: 'A rare delicacy.',
+      image: '',
+      lifespan: 4,
+      location: 'Aisle 1',
+      notes: 'For testing only',
+      store: 'Pomme de Terre',
+      tags: [],
+      threshold: 4
+    };
+
+    //First, add a new product to the product list (so we know it's not already in the shopping list)
+    cy.visit('/products/new');
+
+    page.getAddProductFormField('productName').type(testNewProduct.productName);
```

Comment:
I wonder if these next several lines (and maybe everything from visiting `/products/new` down to `addProductSubmitButton()`) could/should be in a PO file. You could then have a single function call here that's something like "addNewProduct(testNewProduct)`?

Reply:
The comment on line 263 suggests that there is a function trying to escape here.